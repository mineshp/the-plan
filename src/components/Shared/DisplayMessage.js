import React from 'react';
import { Message } from 'semantic-ui-react';
import PropTypes from 'prop-types';

const DisplayMessage = (props) => {
    let isError = false;

    // TODO check other success and error messages
    if (props.status.error && props.status.error.isError) {
        isError = props.status.error.isError;
    }

    const color = isError
        ? 'red'
        : 'olive';

    const message = (props.status.error && props.status.error.message)
        ? props.status.error.message
        : props.status.success.message;

    return (
        <Message color={color}>
            <p>{message}</p>
        </Message>

    );
};

// Take one object with a messageType field ['success', 'error']
DisplayMessage.propTypes = {
    status: PropTypes.shape({
        error: PropTypes.shape({
            isError: PropTypes.bool,
            message: PropTypes.string
        }),
        success: PropTypes.shape({
            message: PropTypes.string
        })
    }).isRequired
};

export default DisplayMessage;
