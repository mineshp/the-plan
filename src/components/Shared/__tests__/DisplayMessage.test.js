import React from 'react';
import renderer from 'react-test-renderer';
import DisplayMessage from '../DisplayMessage';

it('renders a success message', () => {
    const successfulResult = {
        success: {
            message: 'Successful request',
        }
    };

    const tree = renderer.create(<DisplayMessage status={successfulResult} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.props.className).toContain('olive');
});

it('renders an error message', () => {
    const errorResult = {
        error: {
            message: 'Error request',
            isError: true
        }
    };

    const tree = renderer.create(<DisplayMessage status={errorResult} />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree.props.className).toContain('red');
});
