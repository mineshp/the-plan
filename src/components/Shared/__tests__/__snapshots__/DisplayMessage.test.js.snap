// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayMessage renders a success message 1`] = `
<div
  className="ui olive message"
>
  <p>
    Successful request
  </p>
</div>
`;

exports[`DisplayMessage renders an error message 1`] = `
<div
  className="ui red message"
>
  <p>
    Error request
  </p>
</div>
`;
